### DISCUSSION TITLE
Additional CSS at site level - how is it done?

### DISCUSSION THREAD

**Discussion by dajare** [datenstrom/community#699] [Ask a question] [answered]
**Scenario**: I wish to **add** a CSS file to my Yellow site. I called it `add.css`, and put it in the `system/themes` folder. I *thought* I could simply insert

    <link rel="stylesheet" type="text/css" href="/yellow/system/layouts/add.css" />

to the `header.html` file, but that 404s. I have also tried making a "layout" file called `add.html` with the code in it, and including that in `header.html` with 

    <?php $this->yellow->layout("add"); ?>

But no matter where I seem to put `add.css` (I have tried several locations!), or whether I used absolute file address or not, it is never found.

Clearly I'm misunderstanding quite a lot! :/ There may well be more than one way to do this, but how does one point to a CSS file "properly"—the Yellow way(!)?

I guess I have two distinct questions:

- how to point at a file?
- how *best* in Yellow terms to include an **additional** CSS file?

Thanks for your patience, and any responses!

(Background: I'm still getting used to Yellow's API, having worked with a very different PHP-based CMS over many years. Of course, I (wrongly!) keep looking for analogies.... I have read the API page over and over, but it isn't always clear to me what some of those examples *return*.  My limitation!)

**Comment by schulle4u** [comment, marked as answer]
I suggest to use the `CoreThemeLocation` setting, it will always use the right path to include  your custom theme files. Using your `add.css` example, you can include the following in your `header.html`. 

```
<link rel="stylesheet" type="text/css" href="<?php echo $this->yellow->system->get("coreServerBase").$this->yellow->system->get("coreThemeLocation")."add.css"; ?>" />
```

You can also write a small extension to avoid modifying layout files. Have a look at [the helloworld extension](https://github.com/schulle4u/yellow-extension-helloworld), it shows how to easily include custom js and css to extend your website.

**Reply by dajare** [reply]
This is what I was looking for, and I see now that the "API" only exposes you to so much, and that the "analogies" I refer to in my original post can be (mostly!) found by scanning `core.php`.

Thanks for the continued assistance. One of these days I'll achieve clarity on how Yellow's bits fit together!

**Comment by IngoBelka** [comment]
Another way could be, to use the css include rule.
`@import "add.css" `

It might be worth a try.

**Reply by dajare** [reply]
I thought about that, but wasn't sure which "master" CSS file to add that to! :man_shrugging: Thanks, though!
